### PR TITLE
Updated README and added missing API doc

### PR DESCRIPTION
**Pull request by helenmasters**

### CODE REVIEW

File: Sources/CloudEnvironment/MongoDBCredentials.swift [hunk 1]
```diff
@@ -16,16 +16,24 @@
 
 import Foundation
 
-/// MongoDBCredentials class
+/// Contains the credentials for a MongoDB service instance. You will typically
+/// receive an instance of this type through `cloudEnv.getMongoDBCredentials(name: String)`.
 ///
-/// Contains the credentials for a MongoDB service instance.
+/// Reference [MongoDB](https://console.ng.bluemix.net/docs/services/MongoDB/index.html).
```
Comment (christiancompton) [inline Sources/CloudEnvironment/MongoDBCredentials.swift]: This reference links to the docs for the service, and not the catalog page itself.  Is this intentional?

File: Sources/CloudEnvironment/ObjectStorageCredentials.swift [hunk 2]
```diff
@@ -14,16 +14,22 @@
 * limitations under the License.
 */
 
-/// ObjectStorageCredentials class
+/// Contains the credentials for an Object Storage service instance. You will typically
+/// receive an instance of this type through `cloudEnv.getObjectStorageCredentials(name: String)`.
 ///
-/// Contains the credentials for an Object Storage service instance.
+/// Reference [Object Storage](https://console.ng.bluemix.net/catalog/services/cloud-object-storage).
```
Comment (christiancompton) [inline Sources/CloudEnvironment/ObjectStorageCredentials.swift]: This Object Storage Service is for the legacy **OpenStack Swift** object storage.  It was removed from the catalog a while ago, but is still an industry standard for Storage APIs.  The new Cloud Object Storage service uses the Amazon S3 API, and will be a completely separate credential set with another class.  

We could either link to [legacy documentation](https://cloud.ibm.com/docs/infrastructure/objectstorage-swift?topic=objectstorage-swift-GettingStarted#getting-started-with-object-storage-openstack-swift) or official [OpenStack Swift docs](https://docs.openstack.org/swift/latest/).  I am not sure how much longer the legacy documentation will be around.

File: Sources/CloudEnvironment/CloudFunctionsCredentials.swift [hunk 3]
```diff
@@ -16,21 +16,22 @@
 
 import Foundation
 
-/// CloudFunctionsCredentials class
+/// Contains the credentials for a Cloud Functions instance. You will typically
+/// receive an instance of this type through `cloudEnv.getCloudFunctionsCredentials(name: String)`.
 ///
-/// Contains the credentials for a Cloud Functions instance.
+/// Reference [Cloud Functions](https://cloud.ibm.com/docs/openwhisk?topic=cloud-functions-index#getting-started-with-cloud-functions).
```
Comment (christiancompton) [inline Sources/CloudEnvironment/CloudFunctionsCredentials.swift]: Here we link to the docs, but given that most other places link to the catalog services, we should link to https://cloud.ibm.com/openwhisk

**Review by christiancompton — COMMENTED** [review]
These changes will be very useful to users looking to leverage the reference documentation.  I left three specific comments about a couple services.

In most of the *Reference [Service]* lines, all instances `console.ng.bluemix.net/...` urls need to be replaced with `cloud.ibm.com/...`.

**Review by christiancompton — APPROVED** [review]

File: Sources/CloudEnvironment/ObjectStorageCredentials.swift (hunk 2 above)
Comment (helenmasters) [reply Sources/CloudEnvironment/ObjectStorageCredentials.swift]: I've linked to the official OpenStack docs and added a note about IBM Cloud, please do check the wording.

File: Sources/CloudEnvironment/CloudFunctionsCredentials.swift (hunk 3 above)
Comment (helenmasters) [reply Sources/CloudEnvironment/CloudFunctionsCredentials.swift]: I've implemented this in the latest commit.

File: Sources/CloudEnvironment/MongoDBCredentials.swift (hunk 1 above)
Comment (helenmasters) [reply Sources/CloudEnvironment/MongoDBCredentials.swift]: I've changed all the links for services to link consistently to the catalog service pages as you suggested.

**Pull request merged**